src/tests/unit/tests/background/actions/details-view-action-creator.test.ts

```diff
@@ -0,0 +1,120 @@
+import { DetailsViewController } from '../../../../../background/details-view-controller';
+import { TelemetryEventHandler } from '../../../../../background/telemetry/telemetry-event-handler';
+import { Action } from '../../../../../common/flux/action';
+import { RegisterTypeToPayloadCallback } from '../../../../../common/message';
+import { Messages } from '../../../../../common/messages';
+import { SETTINGS_PANEL_CLOSE, SETTINGS_PANEL_OPEN, TelemetryEventSource, TriggeredBy } from '../../../../../common/telemetry-events';
+import { DetailsViewRightContentPanelType } from '../../../../../DetailsView/components/left-nav/details-view-right-content-panel-type';
+
+describe('DetailsViewActionCreatorTest', () => {
+    let registerCallbackMock: IMock<RegisterTypeToPayloadCallback>;
+    let detailsViewActionsMock: IMock<DetailsViewActions>;
+    let detailsViewControllerMock: IMock<DetailsViewController>;
+    let telemetryEventHandlerMock: IMock<TelemetryEventHandler>;
+
+    let testSubject: DetailsViewActionCreator;
+
+    beforeEach(() => {
+        registerCallbackMock = Mock.ofType<RegisterTypeToPayloadCallback>();
+        detailsViewActionsMock = Mock.ofType<DetailsViewActions>();
+
+        detailsViewControllerMock = Mock.ofType<DetailsViewController>();
+        telemetryEventHandlerMock = Mock.ofType<TelemetryEventHandler>();
+
+        testSubject = new DetailsViewActionCreator(
+            registerCallbackMock.object,
+            detailsViewActionsMock.object,
+            detailsViewControllerMock.object,
+            telemetryEventHandlerMock.object,
+        );
+    });
+
+    test('on SettingsPanel.OpenPanel', () => {
+        const openSettingsPanelMock = Mock.ofType<Action<null>>();
+
+        detailsViewActionsMock.setup(actions => actions['openSettingsPanel']).returns(() => openSettingsPanelMock.object);
+
+        const payload: BaseActionPayload = {
+            telemetry: {
+                triggeredBy: 'test' as TriggeredBy,
+                source: -1 as TelemetryEventSource,
+            },
+        };
+
+        const tabId = -1;
+
+        registerCallbackMock
+            .setup(register => register(Messages.SettingsPanel.OpenPanel, It.is(isFunction)))
+            .callback((message, listener) => listener(payload, tabId));
+
+        testSubject.registerCallback();
+
+        openSettingsPanelMock.verify(action => action.invoke(null), Times.once());
+        detailsViewControllerMock.verify(controller => controller.showDetailsView(tabId), Times.once());
+        telemetryEventHandlerMock.verify(handler => handler.publishTelemetry(SETTINGS_PANEL_OPEN, payload), Times.once());
+    });
+
+    test('on SettingsPanel.ClosePanel', () => {
+        const closeSeetingsPanelMock = Mock.ofType<Action<null>>();
+
+        detailsViewActionsMock.setup(actions => actions['closeSettingsPanel']).returns(() => closeSeetingsPanelMock.object);
+
+        const payload: BaseActionPayload = {
```

Comment:
Seems to be the same payload as above; maybe you can move this to beforeEach.

Reply:
Updated.